wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift

```diff
@@ -0,0 +1,140 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireDataModel
+import WireSyncEngine
+
+final class DeviceDetailsViewActionsHandler: DeviceDetailsViewActions, ObservableObject {
+
+    var userClient: UserClient
+    var clientRemovalObserver: ClientRemovalObserver?
+    var credentials: ZMEmailCredentials?
+    var certificate: E2eIdentityCertificate?
+    var isProcessing: ((Bool) -> Void)?
+
+    let userSession: UserSession
+
+    init(
+        userClient: UserClient,
+        userSession: UserSession,
+        credentials: ZMEmailCredentials?
+    ) {
+        self.userClient = userClient
+        self.credentials = credentials
+        self.userSession = userSession
+    }
+
+    func fetchCertificate() async -> E2eIdentityCertificate? {
+        do {
+            return try await userClient.fetchE2eIdentityCertificate(e2eIdentityProvider: e2eIdentityProvider())
```

Comment:
I'm confused here. I wouldn't expect to create `e2eIdentityProvider` here.

Reply:
Moved it to `ClientViewController` to be supplied to `DeviceDetailsActionsHandler`